src/docfx/build/xref/InternalXrefSpec.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal class InternalXrefSpec : XrefSpec
+    {
+        public new Dictionary<string, Lazy<Func<string, string, Document, Document, JValue>>> ExtensionData { get; } = new Dictionary<string, Lazy<Func<string, string, Document, Document, JValue>>>();
```

Comment:
avoid inheritance, `new XXX ExtensionData` makes it harder to reason about base class logic and deserialization/serialization behavior.

Reply:
`InternalXrefSpec` is not used for de-serialization/serialization, but sure to clarify the logic we can rename it to something else.